File: leetcode-submissions/2501_Longest_Square_Streak_in_an_Array.cpp
class Solution {
public:
    int longestSquareStreak(vector<int>& nums) {
     // we need to maximize the streak length after sorting the array 
     sort(nums.begin(),nums.end());
     //after that we need to create a map for storing an element with their streak 
     unordered_map<int,int>mp;
     int maxStreak=0;
     for(int i=0;i<nums.size();i++){
        int root=(int)sqrt(nums[i]);
        if( root*root==nums[i] and mp.find(root)!=mp.end()){
            //then sqrt is present then we need to add 
            mp[nums[i]]=mp[root]+1;
            
        }else{

            mp[nums[i]]=1;
        }

        maxStreak=max(maxStreak,mp[nums[i]]);
     }   
     //after finding the streaks 
    
     return maxStreak<2?-1:maxStreak;
    }
};

File: leetcode-submissions/2501_Longest_Square_Streak_in_an_Array.ts
function longestSquareStreak(nums: number[]): number {
    let maxStreak=0;
    const mp=new Map<number ,number >();
    nums.sort((a,b)=>a-b);
    for(const n of nums ){
        //in this we need to find out the sqrt 
        const root:number =Number(Math.sqrt(n));
        if(root*root===n && mp.has(root)){
            mp.set(n,mp.get(root)+1);
        }else{
            mp.set(n,1);
        }
        maxStreak=Math.max(maxStreak,mp.get(n));
    }
    return maxStreak<2 ?-1:maxStreak;
};